python/cli/process/process.py

```diff
@@ -485,6 +476,23 @@ def write_colmap_csv(data, fn):
 
             finalMapWritten = True
 
+    def onVioOutput(vioOutput):
+        nonlocal visualizer, isTracking
+        wasTracking = isTracking
+        isTracking = vioOutput.status == spectacularAI.TrackingStatus.TRACKING
+        if wasTracking and not isTracking:
+            print('warning: Lost tracking!')
+
+        if visualizer is not None:
+            visualizer.onVioOutput(vioOutput.getCameraPose(0), status=vioOutput.status)
+
+    def onMappingOutput(output):
+        try:
+            processMappingOutput(output)
+        except Exception as e:
+            print(f"Processing mapping output failed: {e}", flush=True)
```

Comment:
Should the error message be more clear that this shouldn't happen and suggest users to report the crash to us at Github?

Reply:
Does this work if you put the catch to the `if __name__ == '__main__':` block, or do you have to catch it here?

As a general principle, the system should never crash and we do not have to decorate these error messages. It could be just `ERROR: {e}`

Reply:
I now tried having `try ... expect ...` block there but it didn't work. Probably related to mapping output being called from different thread.